Orange/widgets/unsupervised/owkmeans.py

```diff
@@ -27,355 +67,254 @@ class OWKMeans(widget.OWWidget):
-            callback=self.update)
+        bg = gui.radioButtonsInBox(
+            self.controlArea, self, "optimize_k", orientation=layout,
+            box="Number of Clusters", callback=self.apply)
         layout.addWidget(
-            gui.appendRadioButton(bg, "Fixed:", addToLayout=False),
-            1, 1)
+            gui.appendRadioButton(bg, "Fixed:", addToLayout=False), 1, 1)
         sb = gui.hBox(None, margin=0)
-        self.fixedSpinBox = gui.spin(
+        gui.spin(
             sb, self, "k", minv=2, maxv=30,
             controlWidth=60, alignment=Qt.AlignRight, callback=self.update_k)
         gui.rubber(sb)
         layout.addWidget(sb, 1, 2)
 
         layout.addWidget(
-            gui.appendRadioButton(bg, "Optimized from", addToLayout=False), 2, 1)
+            gui.appendRadioButton(bg, "From", addToLayout=False), 2, 1)
         ftobox = gui.hBox(None)
         ftobox.layout().setContentsMargins(0, 0, 0, 0)
-        layout.addWidget(ftobox)
+        layout.addWidget(ftobox, 2, 2)
         gui.spin(
             ftobox, self, "k_from", minv=2, maxv=29,
             controlWidth=60, alignment=Qt.AlignRight,
             callback=self.update_from)
         gui.widgetLabel(ftobox, "to")
-        self.fixedSpinBox = gui.spin(
+        gui.spin(
             ftobox, self, "k_to", minv=3, maxv=30,
             controlWidth=60, alignment=Qt.AlignRight,
             callback=self.update_to)
         gui.rubber(ftobox)
 
-        layout.addWidget(gui.widgetLabel(None, "Scoring: "),
-                         5, 1, Qt.AlignRight)
-        layout.addWidget(
-            gui.comboBox(
-                None, self, "scoring", label="Scoring",
-                items=list(zip(*self.SCORING_METHODS))[0],
-                callback=self.update), 5, 2)
-
         box = gui.vBox(self.controlArea, "Initialization")
         gui.comboBox(
             box, self, "smart_init", items=self.INIT_METHODS,
-            callback=self.update)
+            callback=self.invalidate)
 
         layout = QGridLayout()
-        box2 = gui.widgetBox(box, orientation=layout)
-        box2.setSizePolicy(QSizePolicy.Maximum, QSizePolicy.Maximum)
-        layout.addWidget(gui.widgetLabel(None, "Re-runs: "),
-                         0, 0, Qt.AlignLeft)
+        gui.widgetBox(box, orientation=layout)
+        layout.addWidget(gui.widgetLabel(None, "Re-runs: "), 0, 0, Qt.AlignLeft)
         sb = gui.hBox(None, margin=0)
         layout.addWidget(sb, 0, 1)
         gui.lineEdit(
             sb, self, "n_init", controlWidth=60,
-            valueType=int, validator=QIntValidator(),
-            callback=self.update)
-        layout.addWidget(gui.widgetLabel(None, "Maximal iterations: "),
-                         1, 0, Qt.AlignLeft)
+            valueType=int, validator=QIntValidator(), callback=self.invalidate)
+        layout.addWidget(
+            gui.widgetLabel(None, "Maximal iterations: "), 1, 0, Qt.AlignLeft)
```

Comment:
This was not changed and probably does not need fixing (maximal should be correct in this use), but I think "Maximum iterations" is a bit more standard...